File: src/main/java/com/example/finances/repositories/UserRepository.java
package com.example.finances.repositories;

import com.example.finances.domains.user.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Integer> {

    Optional<User> findById(Integer id);
    Optional<User> findByEmail(String email);
}

File: src/main/kotlin/com/example/finances/repositories/UserRepository.kt
package com.example.finances.repositories

import com.example.finances.user.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): Optional<User>

    @Modifying(clearAutomatically = true, flushAutomatically = true)
    @Query(value = "UPDATE users SET password = :pwd WHERE id = :id", nativeQuery = true)
    fun updatePasswordById(@Param("id") id: Long, @Param("pwd") pwd: String): Int
}
